### DISCUSSION TITLE
Rendering sortable-gallery with next.js Image component does not show overlay

### DISCUSSION THREAD

**Discussion by park-onezero** [igordanchenko/react-photo-album#180] [Q&A] [answered]
I combined the two examples, but there is a problem that the overlay image does not appear and the image blinks when onDrop.

![renderNextImage 적용 전](https://github.com/user-attachments/assets/5e6b202d-76e8-4ec1-85a0-ef108f8caa09)
*without renderNextImage*

![renderNextImage 적용 후](https://github.com/user-attachments/assets/1ca69c54-23e1-42cf-a14c-e83125f4a242)
*with renderNextImage*

```tsx
function renderNextImage({ alt = "", title, sizes }: RenderImageProps, { photo, width, height }: RenderImageContext) {
  return (
    <div
      style={{
        width: "100%",
        position: "relative",
        aspectRatio: `${width} / ${height}`,
      }}
    >
      <Image
        fill
        src={photo}
        alt={alt}
        title={title}
        sizes={sizes}
        placeholder={"blurDataURL" in photo ? "blur" : undefined}
      />
    </div>
  );
}

<SortableGallery
  render={{
    image: renderNextImage,
  }}
  gallery={RowsPhotoAlbum}
  spacing={4}
  photos={photos}
  movePhoto={(oldIndex, newIndex) => setPhotos(arrayMove(photos, oldIndex, newIndex))}
/>
```

**Comment by igordanchenko** [comment, marked as answer]
Thank you for bringing this up. This is happening because `SortableGallery` example is unaware of the DOM structure of photos rendered with a custom render function.

You should be able to make the two examples work together with the following modifications:

1) Add `data-id` attribute to the `next/image` image:

```jsx
<Image
 data-id={photo.src}
 // ...
/>
```

2) Modify `handleDragStart` function in the `SortableGallery.tsx`:

```jsx
const image = ref.current?.querySelector(`img[data-id="${photo?.id}"]`);
const padding = image?.parentElement?.parentElement
  ? getComputedStyle(image.parentElement.parentElement).padding
  : undefined;
```

**Reply by park-onezero** [reply]
Thank you. It works well!
